### PR TITLE
HAL_ChibiOS: reduce rebuild on re-configure

### PR DESCRIPTION
**Pull request by tridge**
this fixes an unnecessary change in hwdef.h based on set to list conversion, and avoids changing timestamp on hwdef.h if unchanged

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 1]
```diff
@@ -2222,6 +2224,21 @@ def write_hwdef_header(outfilename):
...
+        if filecmp.cmp(outfilename, tmpfile):
+            print("No change in hwdef.h")
+            os.unlink(tmpfile)
+            return
+    except FileNotFoundError:
+        pass
+    try:
+        os.unlink(outfilename)
+    except FileNotFoundError:
+        pass
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2239]:
```suggestion
```

`os.rename` quite happily blats the second file.

**Comment by peterbarker** [comment]
Works fine for me locally.

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2239]: you tested on windows? on cygwin? on macos?
better to handle the error than just guess

**Comment by peterbarker** [comment]
Success reported on Discord, merged!

**Pull request merged**